app/services/api/v1/ecf_participants/index.rb

```diff
@@ -4,53 +4,60 @@ module Api
   module V1
     module ECFParticipants
       class Index
-        def initialize(lead_provider, params)
-          self.lead_provider = lead_provider
-          self.params        = params
+        attr_reader :cpd_lead_provider, :params
+
+        def initialize(cpd_lead_provider:, params:)
+          @cpd_lead_provider = cpd_lead_provider
+          @params = params
         end
 
         def induction_records
-          join = InductionRecord
-                   .select("induction_records.id, ROW_NUMBER() OVER (PARTITION BY induction_records.participant_profile_id ORDER BY induction_records.created_at DESC) AS created_at_precedence")
-                   .joins(:participant_profile, :schedule, { induction_programme: :partnership })
-                   .where(
-                     schedule: { cohort_id: with_cohorts.map(&:id) },
-                     induction_programme: {
-                       partnerships: {
-                         lead_provider_id: lead_provider.id,
-                         challenged_at: nil,
-                         challenge_reason: nil,
-                       },
-                     },
-                   )
-
           scope = InductionRecord
-                    .references(participant_profile: %i[participant_identity])
-                    .includes(
-                      :preferred_identity,
-                      :schedule,
-                      induction_programme: { school_cohort: %i[school cohort] },
-                      mentor_profile: [:participant_identity],
-                      participant_profile: %i[ecf_participant_eligibility ecf_participant_validation_data participant_identity teacher_profile user],
-                    )
-                    .joins("JOIN (#{join.to_sql}) AS latest_induction_records ON latest_induction_records.id = induction_records.id AND latest_induction_records.created_at_precedence = 1")
+            .where(id: induction_record_ids_with_deduped_profiles)
+            .joins(schedule: [:cohort], participant_profile: { school_cohort: [:cohort] })
+            .where(schedule: { cohort: with_cohorts })
```

Comment:
out of curiosity when would the `schedule.cohort` be different than `participant_profile . school_cohorts .cohort`?

Reply:
The correct way to look the cohort `induction_records.schedule.cohort`. We've seen last week that there are inconsistencies between `participant_profile#schedule` and `induction_record#schedule`

Reply:
That's what you think to be correct, but you're only looking at part of the picture. On the schools side of things they don't care about schedules and hardly look at them. Yet somehow what are able to determine the cohort of a participant. Therefore is `induction_records.schedule.cohort` the correct way to determine's someone cohort?

With regards to `participant_profile#schedule` that is not present in the above snippet therefore I believe has no bearing on this part of the conversation.

Reply:
I've removed the joining and where on `participant_profile` for cohort. This only goes down the schedule part here. elsewhere in this class we also use the cohort from the induction programme. so two routes are used but this one has been removed